packages/utils/src/internals/error_snapshotter.ts

```diff
@@ -0,0 +1,100 @@
+import crypto from 'node:crypto';
+
+import type { CrawlingContext } from '@crawlee/basic';
+import type { KeyValueStore } from '@crawlee/core';
```

Comment:
this package does not depend on core, neither on basic packages, so you cant import from those here. only dependency on utils level is the `@crawlee/types` package, we can move the `CrawlingContext` interface there probably, but we cant do the same with `KeyValueStore`.

either move this to the `core` package (as the `CrawlingContext` is in fact defined on that level, not in the basic package), or introduce new types that match the shape of those, but importing here is not an option.

Reply:
looking at moving the `CrawlingContext` to the types package, it would be also pretty hard to do, so probably better to just move this to the `core` package instead

Reply:
Do you mean import both from core like this?
`import type { KeyValueStore, CrawlingContext } from '@crawlee/core';`

Reply:
@B4nan Do you mean to move the whole class to the core package?

Reply:
yes